### PR TITLE
Adding a new code style rule (IDE0005)

### PR DESCRIPTION
**Pull request by vladimir-krestov**
Rule IDE0005 - "Remove unnecessary import"
It's necessary to have VS warnings, if there are forgotten extra usings in files.

## Proposed changes

- Add a new rule to the `.editorconfig` file

## Customer Impact

- No

## Regression? 

- No

## Risk

- Minimal

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5052)

### CODE REVIEW

**Review by RussKie — DISMISSED** [review]
👍 

Surprised no build failures, we don't have any unused usings :)

**Comment by dreddy-work** [comment]
We need to make sure we pull main before we merge this to avoid build failures for other commits that went in after this fork..

**Comment by vladimir-krestov** [comment]
Rebased to the latest main. Fixed all extra issues. The rule works correctly.
@RussKie, @dreddy-work, please review and merge.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewRowCollectionCodeDomSerializer.cs [hunk 1]
```diff
@@ -44,9 +43,9 @@ protected override object SerializeCollection(IDesignerSerializationManager mana
 
             if (rowCollection.Count > 0)
             {
-                Debug.Assert(rowCollection.Count == 1, " we should have only the add new row");
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewRowCollectionCodeDomSerializer.cs:47]: Why this change? why not keep namespace and keep existing Debug statement? I see other places you had # if DEBUG directive around name space. can you make it consistence?

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/CategoryGridEntry.cs [hunk 2]
```diff
@@ -27,7 +26,7 @@ public CategoryGridEntry(PropertyGrid ownerGrid, GridEntry peParent, string name
 #if DEBUG
             for (int n = 0; n < childGridEntries.Length; n++)
             {
-                Debug.Assert(childGridEntries[n] is not null, "Null item in category subproperty list");
+                Diagnostics.Debug.Assert(childGridEntries[n] is not null, "Null item in category subproperty list");
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/CategoryGridEntry.cs:29]: same here

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewRowCollectionCodeDomSerializer.cs (hunk 1 above)
Comment (vladimir-krestov) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewRowCollectionCodeDomSerializer.cs:47]: I made it usign the following rules:
- If there are 1-2 breaking places after removing an extra `using`, I just add a long class path.
-  If there are 3 or more places, I add the directive

@dreddy-work, should I make the changes similar (add the directive for these places too)?

Comment (dreddy-work) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewRowCollectionCodeDomSerializer.cs:47]: Perf wise it doesn't matter either way. But if these rules are not enforced in the future check-ins for all users in some automatic fashion, its hard to maintain. I think its will be consistent and easy to maintain for we go with #If directive.

**Review by dreddy-work — APPROVED** [review]
